NPE in `invokeHover`

```
Caused by: java.lang.NullPointerException: Cannot invoke "org.aya.core.def.Def.telescope()" because "core" is null
	at org.aya.lsp.actions.ComputeSignature.computeSignature(ComputeSignature.java:40)
	at org.aya.lsp.actions.ComputeSignature.computeSignature(ComputeSignature.java:32)
	at org.aya.lsp.actions.ComputeSignature.invokeHover(ComputeSignature.java:26)
	at org.aya.lsp.server.AyaService.lambda$hover$20(AyaService.java:277)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
```

--- a/lsp/src/main/java/org/aya/lsp/actions/ComputeSignature.java
+++ b/lsp/src/main/java/org/aya/lsp/actions/ComputeSignature.java
@@ -2,11 +2,13 @@
 // Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
 package org.aya.lsp.actions;
 
+import kala.collection.immutable.ImmutableSeq;
 import org.aya.api.distill.DistillerOptions;
 import org.aya.api.ref.DefVar;
 import org.aya.api.ref.LocalVar;
 import org.aya.api.ref.Var;
 import org.aya.cli.library.source.LibrarySource;
+import org.aya.concrete.stmt.Signatured;
 import org.aya.core.def.Def;
 import org.aya.core.term.Term;
 import org.aya.distill.BaseDistiller;
@@ -29,17 +31,22 @@ public interface ComputeSignature {
   static @NotNull Doc computeSignature(@NotNull Var target, boolean withResult) {
     return switch (target) {
       case LocalVar localVar -> BaseDistiller.varDoc(localVar);
-      case DefVar<?, ?> defVar -> computeSignature((Def) defVar.core, withResult);
+      case DefVar<?, ?> defVar -> {
+        @SuppressWarnings("unchecked")
+        var ref = (DefVar<? extends Def, ? extends Signatured>) defVar;
+        // #299: hovering a mouse on a definition whose header is failed to tyck
+        if (ref.core == null && ref.concrete.signature == null) yield Doc.empty();
+        yield computeSignature(Def.defTele(ref), Def.defResult(ref), withResult);
+      }
       default -> Doc.empty();
     };
   }
-
-  static @NotNull Doc computeSignature(@NotNull Def core, boolean withResult) {
+  static @NotNull Doc computeSignature(ImmutableSeq<Term.Param> defTele, Term defResult, boolean withResult) {
     var options = DistillerOptions.pretty();
     var distiller = new CoreDistiller(options);
-    var tele = distiller.visitTele(core.telescope(), core.result(), Term::findUsages);
+    var tele = distiller.visitTele(defTele, defResult, Term::findUsages);
     if (withResult) {
-      var res = core.result().toDoc(options);
+      var res = defResult.toDoc(options);
       if (tele.isEmpty()) return res;
       return Doc.stickySep(tele, Doc.symbol(":"), res);
     } else return tele;

--- a/lsp/src/main/java/org/aya/lsp/utils/Resolver.java
+++ b/lsp/src/main/java/org/aya/lsp/utils/Resolver.java
@@ -27,8 +27,6 @@ import org.aya.util.error.WithPos;
 import org.eclipse.lsp4j.Position;
 import org.jetbrains.annotations.NotNull;
 
-import java.util.Objects;
-
 public interface Resolver {
   /** resolve a symbol by its qualified name in the whole library */
   static @NotNull Option<@NotNull Def> resolveDef(
@@ -110,9 +108,10 @@ public interface Resolver {
     }
 
     @Override public @NotNull Unit visitProj(@NotNull Expr.ProjExpr expr, P param) {
-      if (expr.ix().isRight()) {
+      var field = expr.resolvedIx().get();
+      if (expr.ix().isRight() && field != null) {
         var pos = expr.ix().getRightValue();
-        check(param, Objects.requireNonNull(expr.resolvedIx().get()), pos.sourcePos());
+        check(param, field, pos.sourcePos());
       }
       return StmtConsumer.super.visitProj(expr, param);
     }

